### ISSUE TITLE
Beaconchain requesting old block from forked chain

### ISSUE THREAD

**Issue by stefa2k** [prysmaticlabs/prysm#4052]
```
[2019-11-19 16:03:26]  INFO forkchoice: Executing state transition on block root=0x581c34d9... slot=57477
[2019-11-19 16:03:26]  INFO blockchain: Finished applying state transition attestations=4 deposits=0 slot=57477
[2019-11-19 16:03:29]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:03:33]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:03:37]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:03:41]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:03:45]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:03:49]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:03:50]  INFO forkchoice: Executing state transition on block root=0x498a4884... slot=57479
[2019-11-19 16:03:50]  INFO blockchain: Finished applying state transition attestations=8 deposits=0 slot=57479
...
[2019-11-19 16:07:14]  INFO forkchoice: Executing state transition on block root=0xeda2b698... slot=57496
[2019-11-19 16:07:14]  INFO forkchoice: Starting next epoch epoch=7187 finalizedEpoch=7185 justifiedEpoch=7186 previousJustifiedEpoch=7185
[2019-11-19 16:07:14]  INFO forkchoice: Validator registry information activeValidators=535 averageBalance=3.82205 ETH totalValidators=577
[2019-11-19 16:07:14]  INFO blockchain: Finished applying state transition attestations=4 deposits=0 slot=57496
[2019-11-19 16:07:17]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:07:21]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:07:25]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:07:26]  INFO forkchoice: Executing state transition on block root=0xd0f6e24f... slot=57497
[2019-11-19 16:07:26]  INFO blockchain: Finished applying state transition attestations=4 deposits=0 slot=57497
[2019-11-19 16:07:29]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:07:33]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:07:37]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0
[2019-11-19 16:07:38]  INFO forkchoice: Executing state transition on block root=0xa67d6cf0... slot=57498
[2019-11-19 16:07:38]  INFO blockchain: Finished applying state transition attestations=4 deposits=0 slot=57498
```
A restart of the beacon chain resolved the issue.

Validator looks fine and stakes successfully.

**Comment by torfbolt** [comment]
Same issue for me:
`
[2019-11-19 23:44:21]  INFO sync: Requesting parent block currentSlot=56495 parentRoot=d92eec9b7d3ff1a51880206adef6574a0f37c4f61fc7ec5c511d62116861f6b0`
...

Each attestation of the validator client fails with something like

`[2019-11-19 23:46:20] ERROR validator: Could not request attestation to sign at slot 59791: rpc error: code = Internal desc = Could not process slots up to 59791: expected state.slot 59792 < slot 59791 pubKey=0x80261081c024
`

**Comment by rauljordan** [comment]
Resolved by #4066
